File: look-and-say/look-and-say-java/src/main/java/pl/frognar/_06/Sequence.java
package pl.frognar._06;

public class Sequence {
    public static String lookAndSay(String input) {
        String result = "";
        int count = 1;
        for (int index = 0; index < input.length() - 1; index++) {
            if (input.length() > 1) {
                char currentDigit = input.charAt(index);
                char nextDigit = input.charAt(index + 1);
                if (currentDigit == nextDigit) {
                    count++;
                }
                else {
                    result += "%s%s".formatted(count, currentDigit);
                }
            }
        }

        char lastDigit = input.charAt(input.length() - 1);
        result += "%s%s".formatted(count, lastDigit);
        return result;
    }
}

File: look-and-say/look-and-say-py/04/sequence.py
def look_and_say(input_value: str) -> str:
    result = ''
    count = 1
    for index in range(len(input_value) - 1):
        if input_value[index] == input_value[index + 1]:
            count += 1
        else:
            result += f'{count}{input_value[index]}'
            count = 1
    result += f'{count}{input_value[-1]}'
    return result
